vasern/src/core/vasern-parser/Parser.js

```diff
@@ -253,16 +246,16 @@ const Parser = {
 
   valueTypeToStr: (dataType, value) => {
     let type = dataType;
-    if (dataType.indexOf("#") !== -1) {
+
+    if (dataType.indexOf("#") === 0) {
       type = "ref";
     }
 
-    if (dataType.indexOf("[]") !== -1) {
+    if (dataType.indexOf("[]") === 0) {
```

Comment:
Just another thought: If you check it with "=== 0" it has to be at the first place. But what if the user (as I also did in some cases) wants it to be optional (with a ? in the beginning).
(Or didn't I see something?)

Reply:
Totally forgot that optional notation. I'll make another update to this

Reply:
You could just do
if(dataType.indexOf("?") === 0){ //return string without "?" at first position dataType = dataType.slice(1); }
The question in my opinion is, what do you want to do with optional properties? Because values that are null/undefined already get a value of an empty string before valueTypeToStr() gets called (for example in Document.js).